src/scripts/main.js

```diff
@@ -1,3 +1,24 @@
 'use strict';
 
-// write your code here
+function cloneColumn(columnIndex) {
+  const headRows = table.querySelectorAll('thead tr');
+  const bodyRows = table.querySelectorAll('tbody tr');
+  const footRows = table.querySelectorAll('tfoot tr');
+
+  const insertClonedCell = (rows) => {
+    rows.forEach((row) => {
+      const cells = row.children;
+      const clonedCell = cells[columnIndex].cloneNode(true);
+
+      row.insertBefore(clonedCell, cells[cells.length - 1]);
+    });
+  };
+
+  insertClonedCell(headRows);
+  insertClonedCell(bodyRows);
+  insertClonedCell(footRows);
+}
+
+const table = document.querySelector('table');
```

Comment:
The variable `table` is defined after the `cloneColumn` function, which might cause a reference error if `cloneColumn` is called before `table` is defined. Consider moving the `table` variable declaration before the function definition.